test/functional/p2p_mutated_blocks.py

```diff
@@ -92,5 +97,20 @@ def self_transfer_requested():
         honest_relayer.send_and_ping(block_txn)
         assert_equal(self.nodes[0].getbestblockhash(), block.hash)
 
+        # Check that unexpected-witness mutation check doesn't trigger on a header that doesn't connect to anything
+        assert_equal(len(self.nodes[0].getpeerinfo()), 1)
+        attacker = self.nodes[0].add_p2p_connection(P2PInterface())
+        block_missing_prev = copy.deepcopy(block)
+        block_missing_prev.hashPrevBlock = 123
+        block_missing_prev.solve()
+
+        # Attacker gets a DoS score of 10, not immediately disconnected, so we do it 10 times to get to 100
+        for _ in range(10):
+            assert_equal(len(self.nodes[0].getpeerinfo()), 2)
+            with self.nodes[0].assert_debug_log(expected_msgs=["AcceptBlock FAILED (prev-blk-not-found)"]):
+                attacker.send_message(msg_block(block_missing_prev))
+        attacker.wait_for_disconnect(timeout=5)
```

Comment:
I'm not sure if this is the right place for this test because it has nothing to do with mutated blocks. This test should have existed before my PR, so there is probably a better place for it.

Specifically the property we apparently care about is "a peer sending a disconnected block is assigned 10 DoS points" and a separate "if a peer has 100 DoS points it is disconnected". DoS points should be exposed through getpeerinfo (if they aren't already?) to test this properly which makes any log assertions superfluous.

Reply:
> it has nothing to do with mutated blocks

It kind of does, because we can't tell if the witness is mutated without knowing the height.

Reply:
That said, perhaps `p2p_unrequested_blocks.py` is a better home.

Reply:
I would keep the log reading until there's another way to get the DoS points. It makes it easier to debug this test if if breaks.

Reply:
I just opened a PR to add the `misbehavior_score` to `getpeerinfo` since this seems to be a recurring issue: https://github.com/bitcoin/bitcoin/pull/29530